backend/src/event/events.controller.ts

```diff
@@ -7,34 +7,93 @@ import {
   Param,
   Body,
   UsePipes,
+  Res,
 } from '@nestjs/common';
 import { EventsDto } from './events.dto';
 import { EventsService } from './events.service';
 import { ValidationPipe } from 'validations/validation.pipe';
 import { Events } from './events.entity';
+import { ValidationService } from '../utils/validations/validations.service';
 
 @Controller('events')
 export class EventsController {
-  constructor(private readonly eventsService: EventsService) {}
+  constructor(
+    private readonly eventsService: EventsService,
+    private readonly validationService: ValidationService,
+  ) {}
   @Get()
-  async findAll(): Promise<Events[]> {
-    return this.eventsService.findAll();
+  async findAll(@Res() response): Promise<Events[]> {
+    try {
+      const events = await this.eventsService.findAll();
+      if (events) {
+        return response.status(200).send(events);
+      } else {
+        return response
+          .status(404)
+          .send(`Could not find any events in the system.`);
+      }
+    } catch (error) {
+      return response
+        .status(500)
+        .send(`Failed to get events: ${error.message}`);
+    }
   }
 
   @Post()
   @UsePipes(new ValidationPipe())
-  create(@Body() event: EventsDto) {
-    return this.eventsService.createEvent(event);
+  async create(@Res() response, @Body() event: EventsDto) {
+    try {
+      const createdEvent = await this.eventsService.createEvent(event);
```

Comment:
Would it be also possible to validate the event data before trying to create it?